src/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect } from 'react';
+import {
+	PingType,
+	IsPollingType,
+	GetPollingConfigType,
+	UseOnlineEffectType,
+} from './types';
+
+export const needsPolling: IsPollingType = navigator => {
+	// these browsers don't fully support navigator.onLine, so we need to use a polling backup
+	const unsupportedUserAgentsPattern = /Windows.*Chrome|Windows.*Firefox|Linux.*Chrome/;
+
+	if (
+		typeof navigator !== 'undefined' &&
+		unsupportedUserAgentsPattern.test(navigator.userAgent)
+	) {
+		return true;
+	}
+
+	return false;
+};
+
+export const ping: PingType = ({ url, timeout }) => {
+	return new Promise(resolve => {
+		const isOnline = () => resolve(true);
+		const isOffline = () => resolve(false);
+
+		const xhr = new XMLHttpRequest();
+
+		xhr.onerror = isOffline;
+		xhr.ontimeout = isOffline;
+		xhr.onreadystatechange = () => {
+			if (xhr.readyState === xhr.HEADERS_RECEIVED) {
+				if (xhr.status) {
+					isOnline();
+				} else {
+					isOffline();
+				}
+			}
+		};
+
+		xhr.open('HEAD', url);
+		xhr.timeout = timeout;
+		xhr.send();
+	});
+};
+
+const getPollingConfigs: GetPollingConfigType = (
+	pollingConfig,
+	needsPolling
+) => {
+	const defaultConfig = {
+		enabled: true,
+		url: 'https://ipv4.icanhazip.com/',
+		timeout: 5000,
+		interval: 5000,
+	};
+
+	if (
+		(typeof pollingConfig === 'object' && pollingConfig.enabled === true) ||
+		(needsPolling === true && typeof pollingConfig === 'object')
```

Comment:
In case `pollingObject.enabled === false`, this would return an incorrect object 
```
{
    enabled: false,
    url: 'https://ipv4.icanhazip.com/',
    timeout: 5000,
    interval: 5000,
}
```
Change to `(needsPolling === true && typeof pollingConfig === 'object' && pollingConfig.enabled === undefined)`